docs/06/01/faq.md

```diff
@@ -0,0 +1,177 @@
+
+- You want to maintain visual consistency with your existing React design system
+- You need to use specific UI component libraries that aren't compatible with the SDK's default styling
+- You need to modify the behavior of components beyond what's possible with styling
+
+#### How can I ensure my custom components maintain accessibility features?
+
+When implementing custom components, pay attention to:
+
+- Proper labeling of form controls
+- Appropriate ARIA attributes
+- Keyboard navigation
+- Focus management
+- Color contrast
+
+The prop interfaces include properties like `aria-describedby`, `isInvalid`, and others that support accessibility. Make sure your custom implementations use these props correctly.
+
+The SDK components follow accessibility best practices by default. When creating custom implementations, study the default components to understand how they handle accessibility concerns. The SDK uses [React Aria](https://react-spectrum.adobe.com/react-aria/) for accessible components, which you can reference for implementation details.
+
+### Implementing Components
+
+#### How do I implement components with complex behavior like ComboBox or DatePicker?
+
+For complex components, you have a few options:
+
+1. Use a third-party library that provides similar functionality
+2. Implement a simplified version that meets your specific needs
+3. Use the SDK's default implementation for complex components while customizing simpler ones
+
+For example, to implement a DatePicker with react-datepicker:
+
+```tsx
+import DatePicker from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
+import type { DatePickerProps as SDKDatePickerProps } from '@gusto/embedded-react-sdk'
+
+const MyCustomDatePicker = ({
+  label,
+  description,
+  errorMessage,
+  isRequired,
+  isDisabled,
+  isInvalid,
+  id,
+  name,
+  value,
+  onChange,
+  minDate,
+  maxDate,
+  ...props
+}: SDKDatePickerProps) => {
+  return (
+    <div className="date-field">
+      <label htmlFor={id || name}>
+        {label}
+        {isRequired && <span aria-hidden="true"> *</span>}
+      </label>
+      {description && <div className="description">{description}</div>}
+      <DatePicker
+        id={id || name}
+        selected={value}
+        onChange={date => onChange && onChange(date)}
+        disabled={isDisabled}
+        minDate={minDate}
+        maxDate={maxDate}
+        className={isInvalid ? 'invalid' : ''}
+      />
+      {isInvalid && errorMessage && <div className="error">{errorMessage}</div>}
+    </div>
+  )
+}
+```
+
+To understand the expected behavior of complex components, refer to the SDK's implementations:
+
+- [ComboBox on GitHub](https://github.com/Gusto/embedded-react-sdk/tree/main/src/components/Common/UI/ComboBox)
+- [DatePicker on GitHub](https://github.com/Gusto/embedded-react-sdk/tree/main/src/components/Common/UI/DatePicker)
+
+#### Can I use a different UI framework for my custom components?
```

Comment:
Might be worth bumping this one to the top?

Reply:
Done